### PR TITLE
Internal: Added elements selection state [ED-7518]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: assets/dev/js/editor/document/elements/component.js [hunk 1]
```diff
@@ -1,8 +1,26 @@
 import ComponentBase from 'elementor-api/modules/component-base';
 import * as commands from './commands/';
 import * as commandsInternal from './commands-internal/';
+import { updateEnvironment } from 'elementor-document/elements/utils';
 
 export default class Component extends ComponentBase {
+	__construct( args = {} ) {
+		super.__construct( args );
+
+		elementor.once( 'document:loaded', () => {
```
Comment (StyleShit) [inline assets/dev/js/editor/document/elements/component.js:10]: @matipojo 
Do you think there is a better event for that?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/elements/commands/select-all.js [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
-				// select all document elements, we should convert them into a flatten, single-dimension array.
-				elementor.elementsModel.get( 'elements' ).map( ( element ) => {
-					return elementor.getContainer( element.id );
-				} ),
-			),
+		const containers = this.flattenContainersList(
+			// The selection mechanism keeps selected elements in a single-dimension object. Therefore, In order to
+			// select all document elements, we should convert them into a flatten, single-dimension array.
+			elementor.elementsModel.get( 'elements' ).map( ( element ) => {
+				return elementor.getContainer( element.id );
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply assets/dev/js/editor/document/elements/commands/select-all.js:7]: Done

File: assets/dev/js/editor/document/elements/component.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (StyleShit) [reply assets/dev/js/editor/document/elements/component.js:10]: Done

File: assets/dev/js/editor/document/elements/component.js [hunk 3]
```diff
@@ -1,8 +1,26 @@
...
+			let prevState;
+
+			$e.store.subscribe( () => {
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply assets/dev/js/editor/document/elements/component.js:13]: Done

File: assets/dev/js/editor/document/elements/utils.js [hunk 4]
```diff
@@ -65,3 +67,54 @@ export function removeElementFromDocumentState( elementId, parentId, documentSta
 
 	parent.elements = parent.elements.filter( ( childId ) => childId !== elementId );
 }
+
+export function updateEnvironment() {
+	updatePanelPage();
+	updateSortable();
+	updateNavigator();
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply assets/dev/js/editor/document/elements/utils.js:74]: Done

File: assets/dev/js/editor/document/elements/selectors.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	getElementsIds() {
+		return $e.store.getState( 'document/elements/selection' );
+	},
+
+	getContainers( fallback = null ) {
+		const elementsIds = this.getElementsIds() || [];
+
+		if ( ! elementsIds.length && fallback ) {
+			return Array.isArray( fallback ) ? fallback : [ fallback ];
+		}
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply assets/dev/js/editor/document/elements/selectors.js:11]: Can't

**Pull request closed without merging**